clients/src/main/java/org/apache/kafka/clients/CommonClientConfigs.java

```diff
@@ -77,7 +77,8 @@
     public static final String REQUEST_TIMEOUT_MS_DOC = "The configuration controls the maximum amount of time the client will wait "
                                                          + "for the response of a request. If the response is not received before the timeout "
                                                          + "elapses the client will resend the request if necessary or fail the request if "
-                                                         + "retries are exhausted.";
+                                                         + "retries are exhausted. request.timeout.ms should be larger than replica.lag.time.max.ms "
+                                                         + "to reduce message duplication caused by unnecessary producer retry.";
```

Comment:
Perhaps make it clear that replica.lag.time.max.ms is a broker side config. Also, since this is specific to the producer, could we add it only to producer config?